modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney_controller.rb

```diff
@@ -101,6 +110,15 @@ def header_md5
                                                                     'Authorization').to_json)
         end
 
+        def parse_and_verify_poa_code
+          validate_request!(ClaimsApi::V2::ParamsValidation::PowerOfAttorney)
+
+          poa_code = params.dig('serviceOrganization', 'poaCode')
+          validate_poa_code!(poa_code)
+
+          poa_code
+        end
```

Comment:
Disclaimer :: I'm still sick, so it's very possible that I'm not thinking clearly enough yet.

But I think this has uncovered a problem.  I don't think we're going to want the "appoint individual" and "appoint organization" endpoints to expect the same payload structure.  i.e. I don't think we should have the "appoint individual" endpoint looking for a poa_code that is nested in a `service_organization` attribute.  I think the 2 endpoints should expect different payload structures and that they should also have different validators.  Maybe I'm mistaken, or maybe this was already decided back during the original planning and I'm just forgetting it?  Maybe @SlakrHakr remembers?

Reply:
Ugh, actually now that I look at the form again and rethink this, maybe it is valid to allow them to share the same payload structure?  I mean, the actual section on the form is called, `SECTION III: SERVICE ORGANIZATION INFORMATION`, even though you can specify an individual or an organization in there.   🤦

Reply:
Double-Ugh....I forgot that there are 2 different forms.  
- the [21-22A](https://www.vba.va.gov/pubs/forms/vba-21-22a-are.pdf) for appointing an individual
- the [21-22](https://www.vba.va.gov/pubs/forms/vba-21-22-are.pdf) for appointing an organization

And the 2 forms do have different fields for the "service organization" section.  So maybe it does make sense to have the 2 endpoints have different payload structures (and different validators).

Reply:
these 2 endpoints should definitely have different payload structures. thats part of the reason why they are designed to be 2 different endpoints as opposed to 1 single endpoint.

Reply:
I keep going back and forth on this.  One of the reasons I initially backed off of thinking they should have different payload structures, was because both forms have the same section headings.  So it kind of makes sense to allow the payload top-level attributes to map to the form section headings.  But maybe that's not actually a good idea.  Because these forms can change over time.  Form sections could be removed/added/renamed and then our payload structure isn't going to map to the form sections anymore.  Which could be confusing for consumers in the future.

Maybe we should use V2 as a chance to breakaway from tightly mapping our payload structures to the forms they represent?  Maybe the top-level attributes in our structures should be something more like, `veteranInfo`, `claimantInfo`, `representativeInfo`?